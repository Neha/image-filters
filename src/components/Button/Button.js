import React from "react";
import PropTypes from "prop-types";

const Button = props => {
  return <button onClick={props.onClick}>{props.label}</button>;
};

Button.propTypes = {
  label: PropTypes.string.isRequired
};

Button.defaultProps = {
  label: "Default Label"
};

export default Button;
