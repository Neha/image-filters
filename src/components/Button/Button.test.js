import React from "react";
import { configure } from "enzyme";
import renderer from 'react-test-renderer';

import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { shallow } from "enzyme";
import ReactDOM from "react-dom";
import Button from "./Button";

describe('Button test case', () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render("<Button />", div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot testing', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("displays the label", () => {
    const wrapper = shallow(<Button label="click me" />);
    expect(wrapper.find("button").text()).toEqual("click me");
  });
  
  it("simlulate the click", () => {
    const mockCallBack = jest.fn();
  
    const wrapper = shallow(<Button label="click me" onClick={mockCallBack} />);
    wrapper.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
  
})