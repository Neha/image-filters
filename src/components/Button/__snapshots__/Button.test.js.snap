// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test case Snapshot testing 1`] = `
<button>
  Default Label
</button>
`;
